nni/retiarii/nn/pytorch/api.py

```diff
@@ -279,11 +281,29 @@ def forward(self):
 
     def _translate(self):
         # Will function as a value when used in serializer.
-        return self.candidates[0]
+        return self.access(self.candidates[0])
 
     def __repr__(self):
         return f'ValueChoice({self.candidates}, label={repr(self.label)})'
 
+    def access(self, value):
+        if not self._accessor:
+            return value
+        try:
+            v = value
+            for a in self._accessor:
+                v = v[a]
+        except KeyError:
+            raise KeyError(''.join([f'[{a}]' for a in self._accessor]) + f' does not work on {value}')
+        return v
+
+    def __getitem__(self, item):
+        access = copy.deepcopy(self)
+        access._accessor.append(item)
```

Comment:
why use `append`?

Reply:
The underlying implementation is to clone the current instance, and append item to "accessor", which records all the history getitem calls. For example, when accessor is ``[a, b, c]``, the value choice will return `vc[a][b][c]` where `vc` is the original value choice.

Reply:
wow, very fancy, supporting multi-level indices. but does valuechoice support multi-level choice?

Reply:
I think I checked that. Will throw an exception in case the multi-level choice on any of the candidates does not work.